transports/sftp/src/test/java/org/mule/transport/sftp/SftpMessageReceiverTestCase.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.transport.sftp;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import org.mule.api.MuleContext;
+import org.mule.api.construct.FlowConstruct;
+import org.mule.api.endpoint.EndpointURI;
+import org.mule.api.endpoint.InboundEndpoint;
+import org.mule.api.lifecycle.CreateException;
+import org.mule.tck.junit4.AbstractMuleTestCase;
+import org.mule.tck.size.SmallTest;
+import org.mule.util.concurrent.Latch;
+import org.mule.util.lock.LockProvider;
+import org.mule.util.lock.MuleLockFactory;
+import org.mule.util.lock.SingleServerLockProvider;
+
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Before;
+import org.junit.Test;
+
+@SmallTest
+public class SftpMessageReceiverTestCase extends AbstractMuleTestCase
+{
+
+    private static final String CONNECTOR_NAME = "connector-name";
+    private static final String ENDPOINT_URI = "endpoint-uri";
+    private static final String[] FILE_NAMES = new String[] {"some-file-1", "some-file-2"};
+
+    private MuleLockFactory lockFactory;
+    private LockProvider lockProvider;
+    private TestSftpMessageReceiver receiver;
+
+    @Test
+    public void lockIsReleased() throws Exception
+    {
+        // Simulate two poll cycles
+        receiver.poll();
+        receiver.poll();
+
+        // Each time poll is called a new lock should be created for each file, and released after the file is processed.
+        // If the first created lock is not released the second call to poll won't create a new lock but use the previous one.
+        verify(lockProvider, times(2 * FILE_NAMES.length)).createLock(any(String.class));
+    }
+
+    @Test
+    public void lockContainsEndpointUri() throws Exception
+    {
+        receiver.poll();
+
+        for(String fileName : FILE_NAMES)
+        {
+            verify(lockProvider).createLock(getLockId(fileName));
+        }
+    }
+
+    private String getLockId(String fileName)
+    {
+        return String.format("%s-%s-%s", CONNECTOR_NAME, ENDPOINT_URI, fileName);
```

Comment:
It is not the exact same code. SFTP message receiver contains the endpoint as a field. Here I'm simulating the same behaviour of the class I'm checking. Could make public (or package) the receiver.createLockId or pass the endpoint to it, but I think none of them could be a good approach for a test.